examples/campaign_management/add_complete_campaigns_using_mutate_job.py

```diff
@@ -0,0 +1,536 @@
+
+    return ad_group_criterion_operation
+
+
+def build_ad_group_ad_operations(client, ad_group_operations):
+    """Builds new ad group ad operations.
+
+    Args:
+        client: an initialized GoogleAdsClient instance.
+        ad_group_operations: a list of AdGroupOperation instances.
+
+    Returns: a list of AdGroupAdOperation instances.
+    """
+    return [
+        build_ad_group_ad_operation(client, ad_group_operation) \
+        for ad_group_operation in ad_group_operations]
+
+
+def build_ad_group_ad_operation(client, ad_group_operation):
+    """Builds a new ad group ad operation.
+
+    Args:
+        client: an initialized GoogleAdsClient instance.
+        ad_group_operations: an AdGroupOperation instance.
+
+    Returns: an AdGroupAdOperation instance.
+    """
+    ad_group_ad_operation = client.get_type('AdGroupAdOperation', version='v3')
+    # Creates an ad group ad.
+    ad_group_ad = ad_group_ad_operation.create
+    # Creates the expanded text ad info.
+    text_ad = ad_group_ad.ad.expanded_text_ad
+    text_ad.headline_part1.value = f'Cruise to Mars #{uuid4()}'
+    text_ad.headline_part2.value = 'Best Space Cruise Line'
+    text_ad.description.value = 'Buy your tickets now!'
+    final_url = ad_group_ad.ad.final_urls.add()
+    final_url.value = 'http://www.example.com'
+    ad_group_ad.ad_group.value = ad_group_operation.create.resource_name
+    ad_group_ad.status = client.get_type('AdGroupAdStatusEnum',
+                                         version='v3').PAUSED
+
+    return ad_group_ad_operation
+
+
+def run_mutate_job(mutate_job_service, resource_name):
+    """Runs the mutate job for executing all uploaded mutate job operations.
+
+    Args:
+        mutate_job_service: an instance of the MutateJobService message class.
+        resource_name: a str of a resource name for a mutate job.
+
+    Returns: a google.api_core.operation.Operation instance.
+    """
+    try:
+      response = mutate_job_service.run_mutate_job(resource_name)
+      print(f'Mutate job with resource name "{resource_name}" has been '
+            ' executed.')
+      return response
+    except GoogleAdsException as exception:
+      handle_google_ads_exception(exception)
+
+
+def poll_mutate_job(operations_response):
+    """Polls the server until the mutate job execution finishes.
+
+    Sets the initial poll delay time and the total time to wait before time-out.
+
+    Args:
+        operations_response: a google.api_core.operation.Operation instance.
+    """
+    # operations_response represents a Long-Running Operation or LRO. The class
+    # provides an interface for polling the API to check when the operation is
+    # complete. Below we use the synchronous interface, but there's also an
+    # asynchronous interface that uses the Operation.add_done_callback method.
+    # See: https://googleapis.dev/python/google-api-core/latest/operation.html
+    operations_response.result(timeout=MAX_TOTAL_POLL_INTERVAL_SECONDS)
```

Comment:
How complex would it be to use the async interface here instead of sync?

The main benefit of MutateJobService is to be async so it is likely to confuse the reader that a sync call is being used here.

Reply:
Yes this can be asyncronous, I'll update this statement.

The difference is that the `fetch_and_print_results` function gets called from this function instead of `main`

Reply:
I'm going to work on the other comments first - when I use `add_done_callback` the script exits before printing the results and I'm not quite sure how to use Pythons `asyncio` library to run the function so that it waits for the callback to be called.

Reply:
I posted a change to use the async interface. I added some logic that's not necessary for the example in order to call the result printing function after polling is complete, but I added some comments to clarify. See https://github.com/googleads/google-ads-python/pull/257/commits/1797a977052aa6b3f71e35d1d858e0345a1baf53